retrieve_files.sh

```diff
@@ -0,0 +1,171 @@
+#
+###################################################################################################################
+
+#create lock file
+function lock ( ) {
+  if "$1" == true
+  then
+    if [ -f ${lockFile} ]
+    then
+      echo
+      echo "${lockFile} exits."
+      echo "Looks like a copy of $scriptName is already running... "
+      echo "Abort"
+      log "ERR" " Already running, please check logs for reason... "
+      exit 0
+    else
+      touch ${lockFile}
+    fi
+  else
+    if [ -f $lockFile ] ; then
+      rm $lockFile
+    fi
+  fi
+}
+
+function usage
+{
+    echo "usage: ./retrieve_files.sh [[-d YYYMMDD]| [-a] | [-n]]"
+}
+
+function log()
+{
+  type=$1
+  message=$2
+
+  if [ $type = "WARN" -o $type = "ERR" ]
+  then
+    echo >> $statusDirectory/err-$processingTime
+    echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message" >> $statusDirectory/err-$processingTime
+  fi
+  #
+  # always write into log file
+  #
+  echo
+  echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message"
+  echo >> $statusDirectory/log-$processingTime
+  echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message" >> $statusDirectory/log-$processingTime
+}
+
+#=============================== MAIN BODY OF SCRIPT ===============================
+
+##### Constants
+processingTime=`date +%Y%m%d-%H%M%S`
+scriptName=$0
+statusDirectory=logs
+baseURL="https://bulkdata.uspto.gov/data2/patent/trial/appeal/board/"
+dropLocation="files/`date +%Y%m%d`"
+declare -i fileDate
+retrieveAll=false
+retrieveNone=false
+lockFile="/tmp/file_download.lck"
+
+touch $statusDirectory/log-$processingTime
+
+lock true
+
+case "$1" in
+-d | --date )
+  shift
+  echo "$1"
+  if date -d $1 >/dev/null 2>&1
+  then
+    fileDate=$1
+  else
+    log "ERR" "date passed in is not valid: $1"
+    lock false
+    exit 1
+  fi
+  ;;
+-a | --all )
+  retrieveAll=true
+  ;;
+-n | --none )
+  retrieveNone=true
+  ;;
+-h | --help )
+  usage
+  lock false
+  exit
+  ;;
+* )
+  usage
+  lock false
+  log "ERR" "argument passed in is not valid: $1"
+  exit 1
+esac
+     
+#create directory for downloaded files(if does not exist already)
+mkdir -p $dropLocation
+
+currentdate=$(date +%Y%m%d) 
+
+DATE="`date`"
+log "INFO" "-[JOB START] ${DATE}: ------------"
+log "INFO" "================= Script starting ================="
+
+log "INFO" "Starting file download process"
+
+if $retrieveAll && ! $retrieveNone 
+then
+  wget --directory-prefix=files/`date +%Y%m%d` -e robots=off --cut-dirs=3  --reject="index.html*" --no-parent --recursive --relative --level=1 --no-directories $baseURL
+elif ! $retrieveAll && ! $retrieveNone
+then
+  while [ $fileDate -lt $currentdate ]
+  do
+    friDate=$(date '+%C%y%m%d' -d "$fileDate -$(date -d $fileDate +%u) days + 5 day")
+    year=$(date -d $friDate +%Y) 
+    week=$(date -d $friDate +%V)
+    if [ $year -eq  2015 ]
+    then
+      week=$((10#$week-1))
+      if [ $week -lt 10 ]
```

Comment:
This if clause isn't necessary, as the "2" in `printf "%02d"` indicates that it should only pad the string if it is less than 2 digits.